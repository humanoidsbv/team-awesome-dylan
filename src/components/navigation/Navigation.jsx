import React from 'react';
import PropTypes from 'prop-types';

import styles from './Navigation.module.css';

const Navigation = ({ isMenuVisible }) => (
  <nav
    className={`${styles.navigation} ${
      isMenuVisible ? styles.navigationVisible : ''
    }`}
  >
    <a className={`${styles.item} ${styles.itemActive}`} href="/">
      Timesheets
    </a>
    <a className={styles.item} href="/">
      Team members
    </a>
    <a className={styles.item} href="/">
      Projects
    </a>
    <a className={styles.item} href="/">
      Clients
    </a>
    <a className={styles.item} href="/about.html">
      Documents
    </a>
  </nav>
);

Navigation.defaultProps = {
  isMenuVisible: false,
};

Navigation.propTypes = {
  isMenuVisible: PropTypes.bool
};

export default Navigation;
