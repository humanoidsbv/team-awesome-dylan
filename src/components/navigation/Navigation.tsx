import React from 'react';
import { Link } from 'gatsby';

import styles from './Navigation.module.css';

interface NavigationInterface {
  isMenuVisible: boolean;
}

const Navigation = ({ isMenuVisible }: NavigationInterface): React.ReactElement => (
  <nav
    className={`${styles.navigation} ${
      isMenuVisible ? styles.navigationVisible : ''
    }`}
  >
    <Link className={styles.item} activeClassName={styles.itemActive} to="/">
      Time entries
    </Link>
    <Link
      className={styles.item}
      activeClassName={styles.itemActive}
      to="/team-members"
    >
      Team members
    </Link>
    <a className={styles.item} href="/">
      Projects
    </a>
    <Link
      className={styles.item}
      activeClassName={styles.itemActive}
      to="/clients"
    >
      Clients
    </Link>
    <a className={styles.item} href="/about.html">
      Documents
    </a>
  </nav>
);

export default Navigation;
