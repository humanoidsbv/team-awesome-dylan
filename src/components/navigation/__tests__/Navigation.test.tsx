import * as React from 'react';
import { shallow } from 'enzyme';

import Navigation from '../Navigation';

it('matches snapshot', () => {
  const navigation = shallow(
    <Navigation
      isMenuVisible= {true}
    />
  );
  expect(navigation).toMatchSnapshot();
});
