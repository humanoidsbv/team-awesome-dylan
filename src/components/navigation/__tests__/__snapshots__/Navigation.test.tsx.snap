// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="navigation navigationVisible"
>
  <ForwardRef
    activeClassName="itemActive"
    className="item"
    to="/"
  >
    Time entries
  </ForwardRef>
  <ForwardRef
    activeClassName="itemActive"
    className="item"
    to="/team-members"
  >
    Team members
  </ForwardRef>
  <a
    className="item"
    href="/"
  >
    Projects
  </a>
  <ForwardRef
    activeClassName="itemActive"
    className="item"
    to="/clients"
  >
    Clients
  </ForwardRef>
  <a
    className="item"
    href="/about.html"
  >
    Documents
  </a>
</nav>
`;
